tests/plugins/test_variable_redefinition_in_foreach.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+
+from pathlib import Path
+
+from tests.plugins import PluginTestCase
+from troubadix.plugin import LinterWarning
+from troubadix.plugins.variable_redefinition_in_foreach import (
+    CheckVariableRedefinitionInForeach,
+)
+
+
+class CheckVariableRedefinitionInForeachTestCase(PluginTestCase):
+    def test_ok(self):
+        nasl_file = Path(__file__).parent / "test.nasl"
+        content = (
+            "urls = ['foo', 'bar']\n"
+            "foreach url ( urls ) {}\n"
+            "url1 = 'foo'\n"
+            "foreach url(make_list(url1,'bar'))"
+        )
+        fake_context = self.create_file_plugin_context(
+            nasl_file=nasl_file, file_content=content
+        )
+        plugin = CheckVariableRedefinitionInForeach(fake_context)
+        results = list(plugin.run())
+
+        self.assertEqual(len(results), 0)
+
+    def test_fail(self):
+        nasl_file = Path(__file__).parent / "test.nasl"
+        content = (
+            "url1 = 'foo'\n"
+            "foreach url ( url ) {}\n"
+            "foreach url(make_list(url1,url)) {}\n"
```

Comment:
```suggestion
            'url1 = "foo";\n'
            "foreach url(url) {\n  display(url);\n}\n"
            "foreach url(make_list(url1, url)) {\n  display(url);\n}\n"
```

Similar to previously

Reply:
implemented in 315a91465f4fde6a0c1a597f8e8d98e22b221062

Reply:
LGTM now, thanks a lot for taking care :+1: